menas/src/main/scala/za/co/absa/enceladus/menas/repositories/OozieRepository.scala

```diff
@@ -188,6 +216,60 @@ class OozieRepository @Autowired() (oozieClientRes: Either[OozieConfigurationExc
         classLoader.getResourceAsStream(fileName), "UTF-8")).lines().toArray().mkString("\n")
   }
 
+  /**
+   * This is a helper function for impersonating oozie calls using the proper proxy user if configured
+   * 
+   * @param user User to impersonate
+   * @fn Oozie action to perform - Important to note that this should be Oozie action only (only wrap the call to oozieclient)
+   */
+  private def impersonateWrapper[T](user: String)(fn: () => T) = {
+    if (oozieProxyUser.isEmpty || oozieProxyUserKeytab.isEmpty) {
+      logger.info("Oozie impersonation disabled. missing required configuration parameters 'za.co.absa.enceladus.menas.oozie.proxyUser'" + 
+          " and/or 'za.co.absa.enceladus.menas.oozie.proxyUserKeytab'")
+      fn()
+    } else {
+      //first we login as the proxy user
+      logger.info(s"impersonateWrapper Going to log in as $oozieProxyUser")
+
+      val principal = new PrincipalName(oozieProxyUser, PrincipalName.KRB_NT_PRINCIPAL)
+
+      logger.info(s"impersonateWrapper Creating credentials cache.")
+      val cache = CredentialsCache.create(principal)
+
+      logger.info(s"impersonateWrapper Reading keytab file ${oozieProxyUserKeytab}")
+      val kt = KeyTab.getInstance(new KerberosPrincipal(oozieProxyUser), new java.io.File(oozieProxyUserKeytab))
+      val builder = new KrbAsReqBuilder(principal, kt);
+      val opt = new KDCOptions();
+      opt.set(KDCOptions.RENEWABLE, true)
+      builder.setOptions(opt)
+
+      val realm = Krb5Config.getInstance.getDefaultRealm
+      logger.info(s"impersonateWrapper Using realm ${realm}")
+
+      val sname = PrincipalName.tgsService(realm, realm)
```

Comment:
```suggestion
      val serviceName = PrincipalName.tgsService(realm, realm)
```
I think this is what is meant here?